src/sentry/integrations/request_buffer.py

```diff
@@ -27,29 +27,44 @@ def _convert_obj_to_dict(self, redis_object):
         """
         Convert the request string stored in Redis to a python dict
         """
-
+        redis_object = redis_object.replace("'", '"')
         return json.loads(redis_object)
 
     def _get_all_from_buffer(self, buffer_key):
         """
         Get the list at the buffer key.
         """
 
-        return self.client.lrange(buffer_key, 0, BUFFER_SIZE - 1)
+        ret = []
+        now = datetime.now()
+        for i in reversed(range(BUFFER_SIZE)):
+            with freeze_time(now - timedelta(days=i)):
+                cur = datetime.now().strftime("%Y-%m-%d")
+                buffer_key = self.integrationkey + cur
+                ret.append(self.client.hgetall(buffer_key))
+
+        return ret
 
     def _get_broken_range_from_buffer(self, buffer_key):
         """
         Get the list at the buffer key in the broken range.
         """
 
-        return self.client.lrange(buffer_key, 0, IS_BROKEN_RANGE - 1)
+        ret = []
+        now = datetime.now()
+        for i in reversed(range(IS_BROKEN_RANGE)):
+            cur = (now - timedelta(days=i)).strftime("%Y-%m-%d")
+            buffer_key = f"{self.integrationkey}:{cur}"
+            ret.append(self.client.hgetall(buffer_key))
```

Comment:
You could use a pipeline here, and in _get_all_from_buffer, since you already know all the key names in advance.